grails-app/controllers/au/org/ala/collectory/ProviderCodeController.groovy

```diff
@@ -20,14 +20,15 @@ class ProviderCodeController {
     }
 
     def save() {
-        def providerCodeInstance = new ProviderCode(params)
-        if (!providerCodeInstance.save(flush: true)) {
-            render(view: "create", model: [providerCodeInstance: providerCodeInstance])
-            return
+        ProviderCode.withTransaction {
```

Comment:
Although `withTransaction` works here,  In similar cases like this in other apps, we've been using the `@Transactional` annotation to affected method.  Since the entire content of the mehtod is wrapped with `withTransaction`, I would say it would improve the readability if we were to use the `@Transactional` annotation for this function instead?  https://stackoverflow.com/questions/45953959/difference-between-grails-withtransaction-and-a-transactional-service-method#:~:text=The%20main%20difference%20is%20how,the%20code%20inside%20the%20method. 

I also noticed that other CRUD method in this controller fail with the same error. Please find the comments below.  Please also apply this fix to edit(), update() and delete() methods below.

Reply:
Thanks indeed @sughics I'll do ASAP

Reply:
Actually @vjrj , It appears that this problem exists on CRUD method for other controllers.   ProviderMapController, Institutions, Collections are some of the controllers throwing this error. Could you take a quick look through the controllers and verify that the all of the main the CRUD methods have withTransaction or `@Transactional` annotation where required ?

Reply:
@vjrj Re the message above. I've added the fix for the additional controllers in a PR here: https://github.com/AtlasOfLivingAustralia/collectory/pull/155  . This will be part of the next release i.e. OIDC . Please feel free copy over the changes to non-oidc version if you prefer.